spec/emerald/rubify_spec.rb

```diff
@@ -0,0 +1,131 @@
+      expect(compiled_code).to eq('1.0 - 2.0')
+    end
+
+    it "generates division operations" do
+      compiled_code = Emerald::Rubify.new([Emerald::Atom.new('/'), Emerald::Number.new(1.0), Emerald::Number.new(2.0)]).rubify
+      expect(compiled_code).to eq('1.0 / 2.0')
+    end
+
+    it "generates multiplication" do
+      compiled_code = Emerald::Rubify.new([Emerald::Atom.new('*'), Emerald::Number.new(1.0), Emerald::Number.new(2.0)]).rubify
+      expect(compiled_code).to eq('1.0 * 2.0')
+    end
+
+    it "generates less than operations" do
+      compiled_code = Emerald::Rubify.new([Emerald::Atom.new('<'), Emerald::Number.new(1.0), Emerald::Number.new(2.0)]).rubify
+      expect(compiled_code).to eq('1.0 < 2.0')
+    end
+
+    it "generates greater than operations" do
+      compiled_code = Emerald::Rubify.new([Emerald::Atom.new('>'), Emerald::Number.new(1.0), Emerald::Number.new(2.0)]).rubify
+      expect(compiled_code).to eq('1.0 > 2.0')
+    end
+
+    it "generates less than or equal to operations" do
+      compiled_code = Emerald::Rubify.new([Emerald::Atom.new('<='), Emerald::Number.new(1.0), Emerald::Number.new(2.0)]).rubify
+      expect(compiled_code).to eq('1.0 <= 2.0')
+    end
+
+    it "generates greater than or equal to operations" do
+      compiled_code = Emerald::Rubify.new([Emerald::Atom.new('>='), Emerald::Number.new(1.0), Emerald::Number.new(2.0)]).rubify
+      expect(compiled_code).to eq('1.0 >= 2.0')
+    end
+
+    #it "generates addition with negative arguments" do
+      #compiled_code = Emerald::Rubify.new([Emerald::Atom.new('+'), Emerald::Number.new(1.0), Emerald::Number.new(2.0), Emerald::Number.new(-9)]).rubify
+      #expect(compiled_code).to eq('1.0 + 2.0 - 9')
+    #end
+  end
+
+  describe "generates code for logical operations" do
+    it "generates code for querying nil on a number" do
+      compiled_code = Emerald::Rubify.new([Emerald::Atom.new('nil?'), Emerald::Number.new(1.0)]).rubify
+      expect(compiled_code).to eq('1.0.nil?')
+    end
+
+    it "generates code for querying nil on a string" do
+      compiled_code = Emerald::Rubify.new([Emerald::Atom.new('nil?'), Emerald::String.new('"i like"')]).rubify
+      expect(compiled_code).to eq('"i like".nil?')
+    end
+
+    it "generates code for querying empty on a string" do
+      compiled_code = Emerald::Rubify.new([Emerald::Atom.new('empty?'), Emerald::String.new('"bumblebee"')]).rubify
+      expect(compiled_code).to eq('"bumblebee".empty?')
+    end
+  end
+
+  describe "generates code from lists" do
+    it "generates code from an empty list" do
+      compiled_code = Emerald::Rubify.new([ Emerald::List.new()]).rubify
+      expect(compiled_code).to eq('')
+    end
+
+    it "generates code from a list of numbers" do
+      compiled_code = Emerald::Rubify.new([Emerald::List.new( Emerald::Number.new(3), Emerald::Number.new(2))]).rubify
```

Comment:
This is an odd one, there's a number at the start of a list so it's trying to call a number as a function.

I think the Ruby equivalent of this would be:

```
send(3, 2)
```

Perhaps it should be a compile time error as this could never be correct.